### ISSUE TITLE
Function Score weight returns the wrong reference

### ISSUE THREAD

**Issue by tatarco** [elastic/elasticsearch-java#86]
by the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html#function-weight) a function score can contain only weight and no other inner functions 
in the java api in order to build the FunctionScore we must "hold" a "ContainerBuilder" object, that is only provided as 
a response to exp, gauss etc, the weight method (FunctionScore.java:306) returns builder instead of returning ContainerBuilder making this use case impossible to invoke within the java Api

**Comment by anton-filatov** [comment]
I faced with the same problem. Api specification at [compound.ts](https://github.com/elastic/elasticsearch-specification/blob/98036c318e4eed356e4af22917bb02722234db92/specification/_types/query_dsl/compound.ts#L114-L131) also states that "This container is valid without a variant. Also, despite being documented as a function, 'weight' is actually a container property that can be combined with a function".

**Comment by sunzy** [comment]
```
FunctionScore.Builder builder = new FunctionScore.Builder();
FunctionScore.Builder.ContainerBuilder containerBuilder = builder.new ContainerBuilder();
containerBuilder.filter(f -> f.term(t -> t.field("title").boost(1f).value(FieldValue.of("elastic")))).weight(2d);
```

or use lambda

```
FunctionScore.of(
                                                        fn -> (fn.new ContainerBuilder()).filter(f -> f.term(t -> t.field("title").boost(1f).value(FieldValue.of("elastic")))).weight(2d))
```

**Comment by swallez** [comment]
This has been fixed in PR #257
